src/components/icon-button/icon-button.spec.tsx

```diff
@@ -12,17 +12,79 @@ import StyledIconButton from "./icon-button.style";
 import Icon from "../icon";
 import StyledIcon from "../icon/icon.style";
 import { TooltipProvider } from "../../__internal__/tooltip-provider";
+import Logger from "../../__internal__/utils/logger";
+
+jest.mock("../../__internal__/utils/logger");
 
 describe("IconButton component", () => {
   let wrapper: ReactWrapper;
   let onDismiss: jest.Mock;
+  let loggerSpy: jest.SpyInstance<void, [message: string]> | jest.Mock;
+
+  describe("deprecation warning", () => {
+    beforeEach(() => {
+      loggerSpy = jest.spyOn(Logger, "deprecate");
+      jest.restoreAllMocks();
+    });
+
+    afterEach(() => {
+      loggerSpy.mockRestore();
+    });
+
+    afterAll(() => {
+      loggerSpy.mockClear();
+    });
```

Comment:
Since there is only one test in this `describe` block, you could probably get away with putting the `loggerSpy` logic inside the test itself to reduce the lines of code. I'm happy to accept otherwise though as this is minor :)